### DISCUSSION TITLE
Should impl AsyncRead for std::io::Read?

### DISCUSSION THREAD

**Discussion by infogulch** [tokio-rs/tokio#5225] [Ideas] [answered]
This would allow `Chain`ing `Read`s with `AsyncRead`s. This could be useful when the source data is nontrivial (e.g. more than just a `&[u8]`) but it's generation is compute-bound not io-bound.

**Comment by Darksonn** [comment, marked as answer]
Converting a `std::io::Read` into an `AsyncRead` is quite a complicated process. For example, the `tokio::fs::File` type does this, and its pretty complicated.

**Reply by infogulch** [reply]
Why? I don't think meticulously translating a blocking system api into async is on the table here. I'm just looking for the trivial implementation that entirely ignores Context and Poll::Pending and just passes through any call to poll_read to the underlying impl Read::read.

**Reply by Darksonn** [reply]
We don't provide that because its a major footgun if used with a blocking Read, but you can trivially write a wrapper yourself.

**Reply by infogulch** [reply]
Fair enough.
